acceptance/tests/cli/upgrade_test.go

```diff
@@ -0,0 +1,41 @@
+package cli
+
+import (
+	"testing"
+
+	"github.com/hashicorp/consul-k8s/acceptance/framework/consul"
+)
+
+// TestUpgradeAfterFailedInstall exercises the upgrade command after a failed
+// install. This scenario tests this issue: https://github.com/hashicorp/consul-k8s/issues/1005
+func TestUpgradeAfterFailedInstall(t *testing.T) {
```

Comment:
Generally, acceptance tests should be added for happy case scenarios because they're kind of heavy-handed. These scenarios feel more appropriate as unit tests but I know that we don't have unit tests set up. I've tried mocking it out on this branch, but it still needs more work (and we've synced up on it over zoom): https://github.com/hashicorp/consul-k8s/compare/ihustava/wip-cli-unit-test. 

This is not a blocker but if we do decide to merge it with the acceptance tests, then I think we should at least add a TODO here.

Reply:
Definitely, I will add a TODO here and circle back in the future to remove these tests and cover the behavior in unit tests in the CLI.

Reply:
I'm actually going to remove these tests now and handle unit testing in this PR.